dev-util/pkgconf/pkgconf-1.3.90.recipe

```diff
@@ -0,0 +1,121 @@
+SUMMARY="A lightweight pkg-config replacement"
+DESCRIPTION="A program which helps to configure compiler and linker flags for \
+development libraries. It is similar to pkg-config from freedesktop.org."
+HOMEPAGE="http://pkgconf.org/"
+COPYRIGHT="2011-2017 pkgconf authors"
+LICENSE="ISC"
+REVISION="1"
+SOURCE_URI="https://distfiles.dereferenced.org/pkgconf/pkgconf-$portVersion.tar.xz"
+CHECKSUM_SHA256="112db7f957816901b5fe448715c0a9e38f171d4794c44075b0734a0e7fcfba36"
+PATCHES="pkgconf-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+commandSuffix="$secondaryArchSuffix"
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=""
+fi
+
+libVersion="2.0.0"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	pkgconf$secondaryArchSuffix = $portVersion
+	cmd:pkgconf$secondaryArchSuffix = $portVersion
+	lib:libpkgconf$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	pkgconf${secondaryArchSuffix}_devel = $portVersion
+	devel:libpkgconf$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	pkgconf$secondaryArchSuffix == $portVersion base
+	"
+
+SUMMARY_pkgconfig="$SUMMARY (pkg-config compat symlink)"
+DESCRIPTION_pkgconfig="$DESCRIPTION (pkg-config compatibility symlink)"
+
+PROVIDES_pkgconfig="
+	pkgconf${secondaryArchSuffix}_pkgconfig = $portVersion
+	cmd:pkg_config$secondaryArchSuffix = $portVersion
+	"
+REQUIRES_pkgconfig="
+	pkgconf$secondaryArchSuffix == $portVersion base
+	"
+CONFLICTS_pkgconfig="
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:awk
+	cmd:cmp
+	cmd:diff
+	cmd:gcc$secondaryArchSuffix
+	cmd:grep
+	cmd:make
+	cmd:sed
+	"
+
+TEST_REQUIRES="
+	devel:libatf_c$secondaryArchSuffix
+	"
+TEST_PREREQUIRES="
+	cmd:kyua$commandSuffix
+	"
+BUILD_REQUIRES+="$TEST_REQUIRES"
+BUILD_PREREQUIRES+="$TEST_PREREQUIRES"
+
+defineDebugInfoPackage pkgconf$secondaryArchSuffix \
+	"$binDir/pkgconf" \
+	"$libDir/libpkgconf.so.$libVersion"
+
+BUILD()
+{
+	local pkgConfSubDir="lib/${secondaryArchSuffix/_}/pkgconfig"
+	local pkgConfPath=(
+		"$(finddir B_USER_NONPACKAGED_DEVELOP_DIRECTORY)/$pkgConfSubDir"
+		"$(finddir B_USER_DEVELOP_DIRECTORY)/$pkgConfSubDir"
+		"$(finddir B_SYSTEM_NONPACKAGED_DEVELOP_DIRECTORY)/$pkgConfSubDir"
+		"$(finddir B_SYSTEM_DEVELOP_DIRECTORY)/$pkgConfSubDir"
+		"$(finddir B_USER_NONPACKAGED_DATA_DIRECTORY)/pkgconfig"
+		"$(finddir B_USER_DATA_DIRECTORY)/pkgconfig"
+		"$(finddir B_SYSTEM_NONPACKAGED_DATA_DIRECTORY)/pkgconfig"
+		"$(finddir B_SYSTEM_DATA_DIRECTORY)/pkgconfig"
+		)
+
+	runConfigure ./configure \
+		--with-pkg-config-dir="$(IFS=:; echo \"${pkgConfPath[*]}\")"
```

Comment:
About the errors, it is because `/boot/system/develop/lib/pkgconfig` is passed to `--host` somehow. I'm planning to rework the patches to use `find_path` anyway, so this should not be a problem.

> BTW, I saw that pkgconf 1.4.2 was released very recently.

I noticed it when you suggested some changes weeks ago, but I'm currently lacking the time to work on this. I will see if I could manage to bring the VM up this weekend.

Reply:
I'll test with just one dir:
```
    runConfigure ./configure \
        --with-pkg-config-dir="`finddir B_SYSTEM_DEVELOP_DIRECTORY/$relativeLibDir/pkgconfig"
```
I think this will work.

BTW, I've been thinking about a way to teach pkgconf how to work for both x86 secondary arch and x86_gcc2 primary arch:
Imagine you pass something that would roughly look like:
```
-DHAIKU_SYSTEM_DEVELOP_DIR="`finddir B_SYSTEM_DEVELOP_DIRECTORY`" \
-DHAIKU_RELATIVE_BIN_DIR="$relativeBinDir" \
-DHAIKU_RELATIVE_LIB_DIR="$relativeBinDir" \
```
to the compiler. pkgconf could then examine what `*argv` ends with. pkgconf_x86 would then be usable when building packages for x86_gcc2 primary arch, as well as for x86 secondary arch.
(Of course, with `find_directory()` we would not even need to pass a `HAIKU_SYSTEM_DEVELOP_DIR`.)

End of `*argv` | pkgconfig search path
-- | --
`/bin/pkgconf` | `/system/develop/lib/pkgconfig/`
`/bin/x86/pkgconf` | `/system/develop/lib/x86/pkgconfig/`
`/bin/pkgconf-x86` | `/system/develop/lib/x86/pkgconfig/`

Of course, we would include this after `SECONDARY_ARCHITECTURES="x86"`:
```bash
commandBinDir=$prefix/bin
if [ -n "$secondaryArchSuffix" ]; then
	commandBinDir=$prefix/bin
fi
```
and this in `INSTALL()`:
```bash
	if [ -n "$secondaryArchSuffix" ]; then
		mkdir -p "$commandBinDir"
		ln -r -s "$binDir"/pkgconf "$commandBinDir"
	fi
```
and this after `PROVIDES`:
```bash
if [ -n "$secondaryArchSuffix" ]; then
PROVIDES="$PROVIDES
	cmd:pkgconfig
	"
fi
```

This would be a lot of work... So don't worry if you don't have the time right now.
It can be done later. We could also ignore x86_gcc2, since we know it will be dropped in the future.

Reply:
Ideally no path should be hardcoded at build time, and pkgconf would use find_path to locate the directory. It could even examine the path to determine if a "setarch x86" had been used (and perhpaps we could ease this by providing some extra environment variable in the setarch environment if there is no easy way to detect it).

Anyways, I'm fine with pkgconf pointing to the default (that is x86_gcc2) directory if nothing is specified, and having setarch set the appropriate variables so it ends up using x86 directories when in a setarch environment. Haikuporter could likewise set these by default. I think that is how pkgconf is intended to be used and it is also more generic and allows us to use it for cross-compilation in the future.